nuxeo-features/rest-api/nuxeo-rest-api-server/src/main/java/org/nuxeo/ecm/restapi/server/jaxrs/usermanager/AbstractUMObject.java

```diff
@@ -71,9 +74,13 @@ public Response doDeleteArtifact() {
     }
 
     protected void checkUpdateGuardPreconditions() {
+        checkUpdateGuardPreconditions(currentArtifact);
+    }
+
+    protected void checkUpdateGuardPreconditions(T artifact) {
         NuxeoPrincipal principal = getContext().getCoreSession().getPrincipal();
         if (!principal.isAdministrator()) {
-            if ((!principal.isMemberOf("powerusers")) || !isAPowerUserEditableArtifact()) {
+            if ((!principal.isMemberOf("powerusers")) || !isAPowerUserEditableArtifact(artifact)) {
```

Comment:
SonarLint indicates [here](https://rules.sonarsource.com/java/RSPEC-1066?search=Collapsible%20) i think we can replace it:

```
if (!principal.isAdministrator()
                && (!principal.isMemberOf("powerusers") || !isAPowerUserEditableArtifact(artifact))) {
            throw new WebSecurityException("User is not allowed to edit users");
        }
```

Reply:
I do not want to do any cleanup in the backport here.